elastic/kibana
# [Console] Update extract string literal logic

## Summary

Console's string literal expansion logic has worked quite well for a while using RegExp. RegExp, however, has certain limitations and can be tricky to maintain over time the more complex the RegExp becomes.

This PR revamps the implementation by replacing the regexp with a far more powerful code implementation. This code implementation has the benefit of _understanding_ JSON far better than RegExp ever can and enables us to make it more robust over time. This first version includes a minimal, purpose-built JSON parser that simply extracts JSON string values. This result set should exclude JSON keys because they are not valid targets for triple quote expansion.

Fixes https://github.com/elastic/kibana/issues/72589

Fixes https://github.com/elastic/kibana/issues/72373

### How to test

1. Start Kibana (in OSS should be fine)
2. Navigate to console and send requests who's results will be triple quote expanded. (Might be helpful to insert a document that contains a string value with a backslash in it).

Something like:

```
PUT test1

POST test1/_doc/3
{
  "hello": """
  yes!
  """,
  "array": ["you\ncan\ndo\n\"it"]
}
```

### TODO

- [x] Write a parser specific test

### Checklist

Delete any items that are not applicable to this PR.
